manticore/core/smtlib/visitors.py

```diff
@@ -503,13 +503,31 @@ def visit_BoolAnd(self, expression, *operands):
                         ) == BitVecExtract(operand=value1, offset=beg, size=end - beg + 1)
 
     def visit_BoolNot(self, expression, *operands):
+        """
+        !!a -> a
+        !(a||b) -> !a && !b
+        !(a&&b) -> !a || !b
```

Comment:
Small nit: I think these are out of order as compared to the implementation
```suggestion
        !(a&&b) -> !a || !b
        !(a||b) -> !a && !b
```